Marlin Color UI for STM32F1 (FSMC)

rhapsodyv opened the pull request:
### Description

Democratization of Marlin UIs!!

This is a port of the new Marlin Color UI for STM32F1. Currently FSMC support.

Added support for Chitu and MKS Robin Nano v1.

I will send more two separated PR in the next days. One for SPI (is almost complete) and other that will centralize all TFT IO/XPT for STM32F1: All FSMC and SPI UI will use (LVGL, emulated dogm, color UI).

### Benefits

More users will be able to use the new Marlin Color UI.

### Related Issues

![IMG_9756](https://user-images.githubusercontent.com/81722/89700221-f1e99300-d902-11ea-8d9b-40e828d28172.jpg)

rhapsodyv committed:
port of new marlin ui for STM32F1 (FSMC)

rhapsodyv committed:
new marlin ui support for chitu v5/v6 and mks robin nano v1

rhapsodyv commented:
@thisiskeithb if you dont mind and have some time, can you test it in the nano v1.2? And do a M995 to calibrate the TFT, to send me the calibration values, so I can make fix the default values?
Thanks!

thisiskeithb commented:
> @thisiskeithb if you dont mind and have some time, can you test it in the nano v1.2? And do a M995 to calibrate the TFT, to send me the calibration values, so I can make fix the default values?
> Thanks!

Sure thing. I'm working on another project right now, but I should be able to get to testing this weekend.

rhapsodyv commented:
Current bugfix-2.0x is failing the CI.

rhapsodyv committed:
those tft are 0x8066, that can use the same of ILI9488, so I make it default for that boards

rhapsodyv committed:
smaller buffer size, so the user can add more marlin features

thinkyhead committed:
Update Conditionals_LCD.h

thinkyhead committed:
Update Conditionals_LCD.h

thinkyhead committed:
Don't wrap whole header

thinkyhead committed:
Update pins_CHITU3D_V5.h

thinkyhead committed:
Update pins_CHITU3D_V6.h

thinkyhead committed:
Update pins_MKS_ROBIN_NANO.h

thinkyhead committed:
Patch hex_print

thinkyhead committed:
tweaks

Miraculix200 commented:
![R0038828](https://user-images.githubusercontent.com/29820643/89704847-060baf80-d958-11ea-989b-aef7a152ec6b.JPG)

Will do a calibration now, but I don't think it fixes this problem :)

rhapsodyv commented:
WOW! The screen is mirrored! I will take a look about mirroring in the lcd init code.

In the pins_MKS_NANO_ROBIN.h, try defining all `XPT2046 ` to 0, and try a calibration again.

Miraculix200 commented:
I've commented out the XPT2046 defines in Configuration.h and changed those in pins_MKS_NANO_ROBIN.h to 0, but the touch screen still doesn't react to touch

rhapsodyv commented:
@Miraculix200 can you try this branch? 
https://github.com/rhapsodyv/Marlin/tree/robin-nano-1-color-ui

Miraculix200 commented:
@rhapsodyv Then the display doesn't start (shows the "Booting..." screen). Printer starts however, can be connected through USB. 

I'll try with reset pin comments out. Just a random test

rhapsodyv commented:
> @rhapsodyv I've changed the pins file like in the latest commit of your branch. Then the display doesn't start (shows the "Booting..." screen). Printer starts however, can be connected through USB.
> 
> I'll try with reset pin commented out. Just a random test

Its the TFT_DRIVER I changed to auto. Try setting back to  ILI9488 or ST7796 .

Miraculix200 commented:
Ok. Are you sure TOUCH_BUTTONS_HW_SPI is right? Because this is not an SPI display

rhapsodyv commented:
> Ok. Are you sure TOUCH_BUTTONS_HW_SPI is right? Because this is not an SPI display

Touch is SPI, almost always. Sometimes SW SPI, sometimes HW SPI.

rhapsodyv commented:
I'm testing a Nano v2 now

Miraculix200 commented:
Still doesn't react to touch. I'll try ST7796

rhapsodyv commented:
hahahah I found the problem!!!!
the last @thinkyhead commit removed a "return"!!

Just a moment.

rhapsodyv commented:
Try my branch now. If touch works, we will fix the tft.

Miraculix200 commented:
Well,  it's calibrating itself instead of waiting for me touching it. I'll try again, but I don't think that's right. It gets touch signals from somewhere else

rhapsodyv commented:
Try this branch, I fixed the touch: https://github.com/MarlinFirmware/Marlin/pull/18958

rhapsodyv commented:
> Well, it's calibrating itself instead of waiting for me touching it. I'll try again, but I don't think that's right. It gets touch signals from somewhere else

Try put back it:
```
    #define TOUCH_BUTTONS_HW_SPI
    #define TOUCH_BUTTONS_HW_SPI_DEVICE      2

```

Miraculix200 commented:
Oh I didn't see you changed it to device 1. I still got this: #define TFT_DRIVER ILI9488
#define TFT_BUFFER_SIZE 14400
#define TOUCH_BUTTONS_HW_SPI
#define TOUCH_BUTTONS_HW_SPI_DEVICE 2

I'm not using your branch but looking at your commits and then changing my code. I'll try changing it to 1

rhapsodyv commented:
Ahhh remove that so.

remove it:
```
#define TOUCH_BUTTONS_HW_SPI
#define TOUCH_BUTTONS_HW_SPI_DEVICE 2

```

In my branch I removed. I was trying things... but the problem was the missing `return`

Miraculix200 commented:
Recv: Touch screen calibration completed
Recv: TOUCH_CALIBRATION_X 18797
Recv: TOUCH_CALIBRATION_Y -11823
Recv: TOUCH_OFFSET_X -75
Recv: TOUCH_OFFSET_Y 344
Recv: TOUCH_ORIENTATION TOUCH_LANDSCAPE

rhapsodyv commented:
Thanks! I will put it in the defaults.

Your TFT is mirrored yet?

Miraculix200 commented:
Yes, it's still mirrored.

But when it says "top left", I actually touch the top left corner, not the corner where the cursor is

rhapsodyv commented:
Ok. Give some minute, I will update my branch with a test for the mirror issue.

rhapsodyv commented:
@Miraculix200  try it now in the branch `robin-nano-1-color-ui `

Miraculix200 commented:
That sort of worked. But now it's upside down :D

Miraculix200 commented:
Actually it's now mirrored horizontally AND vertically

rhapsodyv commented:
Just pushed the last one :-)

Miraculix200 commented:
It works now

rhapsodyv commented:
Check if the calibration values changed. I dont think, but is best to check

Miraculix200 commented:
Actually M995 doesn't work anymore now. Says "calibration failed" after touching the 4 corners. I'll compile it again with the values I found during the previous calibration

rhapsodyv commented:
> Actually M995 doesn't work anymore now. Says "calibration failed" after touching the 4 corners. I'll compile it again with the values I found during the previous calibration

Try with a pen. Its very sensible :-/

Miraculix200 commented:
That worked. Got different values now:

Recv: TOUCH_CALIBRATION_X 17880
Recv: TOUCH_CALIBRATION_Y -12234
Recv: TOUCH_OFFSET_X -45
Recv: TOUCH_OFFSET_Y 349
Recv: TOUCH_ORIENTATION TOUCH_LANDSCAPE

rhapsodyv commented:
Thanks!!!
